Meadowlark-js/packages/meadowlark-core/src/model/DocumentIdentity.ts

```diff
@@ -74,6 +75,13 @@ function documentIdentityHashFrom(documentIdentity: DocumentIdentity): string {
  * The resulting Base64Url string is 38 characters long. The first 16 characters are the resource info hash and
  * the remaining 22 characters are the identity hash.
  */
-export function documentIdForDocumentIdentity(resourceInfo: BaseResourceInfo, documentIdentity: DocumentIdentity): string {
-  return `${resourceInfoHashFrom(resourceInfo)}${documentIdentityHashFrom(documentIdentity)}`;
+export function meadowlarkIdForDocumentIdentity(
+  resourceInfo: BaseResourceInfo,
+  documentIdentity: DocumentIdentity,
+): MeadowlarkId {
+  return `${resourceInfoHashFrom(resourceInfo)}${documentIdentityHashFrom(documentIdentity)}` as MeadowlarkId;
+}
+
+export function generateDocumentUuid(): DocumentUuid {
+  return crypto.randomUUID() as DocumentUuid;
```

Comment:
If you run the same POST request twice, then each one is generating a new resource with a different `DocumentUuid`, even though the rest of the stored document is the same.

When doing an UPSERT on an previously stored document, we _must_ keep the existing identifier instead of assigning a new one.

Reply:
We found a problem with the documentUuid, the process created a new uuid each time, but that uuid didn't have a relationship with a document. The code was updated to fix this.